<?xml version="1.0" ?>
<sdf version="1.9">
  <world name="explore_world">
    <scene>
      <ambient>0.4 0.4 0.4 1</ambient>
      <background>0.6 0.8 0.6 1</background>
      <shadows>true</shadows>
    </scene>
    <gravity>0 0 -9.8</gravity>

    <!-- Plugins de mundo -->
    <plugin filename="libignition-gazebo-physics-system.so" name="ignition::gazebo::systems::Physics"/>
    <plugin filename="libignition-gazebo-user-commands-system.so" name="ignition::gazebo::systems::UserCommands"/>
    <plugin filename="libignition-gazebo-scene-broadcaster-system.so" name="ignition::gazebo::systems::SceneBroadcaster"/>
    <plugin filename="libignition-gazebo-sensors-system.so" name="ignition::gazebo::systems::Sensors">
      <render_engine>ogre2</render_engine>
    </plugin>

    <!-- Chão -->
    <model name="ground_plane">
      <static>true</static>
      <link name="link">
        <collision name="collision">
          <geometry><plane><normal>0 0 1</normal><size>50 50</size></plane></geometry>
        </collision>
        <visual name="visual">
          <geometry><plane><normal>0 0 1</normal><size>50 50</size></plane></geometry>
          <material>
            <ambient>0.1 0.4 0.1 1</ambient>
            <diffuse>0.1 0.6 0.1 1</diffuse>
            <specular>0 0 0 1</specular>
          </material>
        </visual>
      </link>
    </model>

    <!-- Obstáculo -->
    <model name="box_obstacle">
      <static>true</static>
      <pose>5 0 0.5 0 0 0</pose>
      <link name="link">
        <collision name="col"><geometry><box><size>0.5 0.5 1.0</size></box></geometry></collision>
        <visual name="vis"><geometry><box><size>0.5 0.5 1.0</size></box></geometry><material><diffuse>1 0 0 1</diffuse></material></visual>
      </link>
    </model>

    <!-- Luz -->
    <light name="sun" type="directional">
      <pose>0 0 10 0 0 0</pose>
      <direction>-0.5 0.1 -1</direction>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <cast_shadows>true</cast_shadows>
    </light>

    <!-- Inclui o modelo com câmera -->
<!-- 
<include>
  <uri>file:///root/workshop_assets/assets/models/vehicle_blue/model.sdf</uri>
  <pose>-3 0 0 0 0 0</pose>
  <name>vehicle_blue</name>
</include>
-->

<include>
  <uri>file:///root/workshop_assets/assets/models/vehicle_blue/model_without_camera.sdf</uri>
  <pose>-3 0 0 0 0 0</pose>
  <name>vehicle_blue</name>
</include>


  </world>
</sdf>

